File: src/main/java/de/cplaiz/activecraftcore/commands/VerifyCommand.java
package de.cplaiz.activecraftcore.commands;

import de.cplaiz.activecraftcore.ActiveCraftPlugin;
import de.cplaiz.activecraftcore.exceptions.ActiveCraftException;
import de.cplaiz.activecraftcore.playermanagement.Profilev2;
import de.cplaiz.activecraftcore.utils.ComparisonType;
import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import java.util.List;

public class VerifyCommand extends ActiveCraftCommand {
    public VerifyCommand(ActiveCraftPlugin plugin) {
        super("verify",  plugin);
    }

    @Override
    public void runCommand(CommandSender sender, Command command, String label, String[] args) throws ActiveCraftException {
        checkPermission(sender);
        checkArgsLength(args, ComparisonType.GREATER_EQUAL, 1);
        Player target = getPlayer(args[0]);
        Profilev2 profile = getProfile(target);
        if (!profile.isDefaultmuted()) {
            sendMessage(sender, this.cmdMsg("not-default-muted"));
            return;
        }
        profile.setDefaultmuted(false);
        messageFormatter.setTarget(profile);
        sendMessage(target, this.cmdMsg("target-message"));
        sendMessage(sender, this.cmdMsg("verify"));
    }

    @Override
    public List<String> onTab(CommandSender sender, Command command, String label, String[] args) {
        return args.length == 1 ? getBukkitPlayernames() : null;
    }
}

File: src/main/java/org/activecraft/activecraftcore/commands/VerifyCommand.kt
package org.activecraft.activecraftcore.commands

import org.activecraft.activecraftcore.ActiveCraftPlugin
import org.activecraft.activecraftcore.exceptions.ActiveCraftException
import org.activecraft.activecraftcore.utils.ComparisonType
import org.bukkit.command.Command
import org.bukkit.command.CommandSender

class VerifyCommand(plugin: ActiveCraftPlugin?) : ActiveCraftCommand("verify", plugin!!) {
    @Throws(ActiveCraftException::class)
    public override fun runCommand(sender: CommandSender, command: Command, label: String, args: Array<String>) {
        assertCommandPermission(sender)
        assertArgsLength(args, ComparisonType.GREATER_EQUAL, 1)
        val target = getPlayer(args[0])
        val profile = getProfile(target)
        if (!profile.isDefaultmuted) {
            sendMessage(sender, this.cmdMsg("not-default-muted"))
            return
        }
        profile.isDefaultmuted = false
        messageFormatter.setTarget(profile)
        sendMessage(target, this.cmdMsg("target-message"))
        sendMessage(sender, this.cmdMsg("verify"))
    }

    public override fun onTab(
        sender: CommandSender,
        command: Command,
        label: String,
        args: Array<String>
    ) = if (args.size == 1) getBukkitPlayernames() else null

}